src/qx/InstructionExecutor.cpp

```diff
@@ -9,7 +9,15 @@ InstructionExecutor::InstructionExecutor(core::QuantumState &s)
 {}
 
 void InstructionExecutor::operator()(Measure const &m) {
-    quantumState.measure(m.qubitIndex, &random::randomZeroOneDouble);
+    quantumState.apply_measure(m.qubitIndex, &random::randomZeroOneDouble);
+}
+
+void InstructionExecutor::operator()(Reset const &r) {
+    quantumState.apply_reset(r.qubitIndex, &random::randomZeroOneDouble);
+}
+
+void InstructionExecutor::operator()(ResetAll const &/* r */) {
```

Comment:
We need the parameter here. At the caller site, you have to invoke InstructionExecutor passing a ResetAll argument. We comment out the name of the argument here to avoid a compilation warning. But we need to specify this function receives a ResetAll argument.